src/opentime/timeRange.h

```diff
@@ -5,99 +5,246 @@
+             */
+            bool contains(TimeRange other) const {
+                return _start_time <= other._start_time && end_time_exclusive() >= other.end_time_exclusive();
+            }
+
+            /**overlaps(other)
+             * params: RationalTime other
+             * self contains other
+             */
+            bool overlaps(RationalTime other) const {
+                return contains(other);
+            }
+
+            /**overlaps(other)
+             * params: TimeRange other
+             * The start of self strictly precedes end of other by a value >= epsilon.
+             * The end of self strictly antecedes start of other by a value >= epsilon.
+             * The converse would be other.overlaps(self)
+             */
+            bool overlaps(TimeRange other, double epsilon = defaultEpsilon) const {
+                double selfStart = _start_time.value() / _start_time.rate();
+                double selfEnd = end_time_exclusive().value() / end_time_exclusive().rate();
+                double otherStart = other._start_time.value() / other._start_time.rate();
+                double otherEnd = other.end_time_exclusive().value() / other.end_time_exclusive().rate();
+//                return (otherEnd - selfEnd >= epsilon) &&
+//                       (otherStart - selfStart >= epsilon) &&
+//                        (selfEnd - otherStart >= epsilon);
+//                return _start_time < other.end_time_exclusive() && other._start_time < end_time_exclusive();
+                return (otherEnd - selfStart >= epsilon) &&
+                       (selfEnd - otherStart >= epsilon);
+            }
+
+            /**before(other, epsilon)
+             * params: TimeRange other
+             * The end of self strictly precedes the start of other by a value >= epsilon.
+             * The converse would be other.before(self)
+             */
+            bool before(TimeRange other, double epsilon = defaultEpsilon) const {
+                double selfEnd = end_time_exclusive().value() / end_time_exclusive().rate();
+                double otherStart = other._start_time.value() / other._start_time.rate();
+                return otherStart - selfEnd >= epsilon;
+            }
+
+            /**before(other, epsilon)
+             * params: RationalTime other
+             * The end of self strictly precedes other by a value >= epsilon.
+             */
+            bool before(RationalTime other, double epsilon = defaultEpsilon) const {
+                double selfEnd = end_time_exclusive().value() / end_time_exclusive().rate();
+                double otherTime = other.value() / other.rate();
+                return otherTime - selfEnd >= epsilon;
+            }
+
+            /**meets(other, epsilon)
+             * params: TimeRange other
+             * The end of self strictly equals the start of other and
+             * the start of self strictly equals the end of other.
```

Comment:
delete this line (start of self does not equal end of other, this line is wrong and unneccesary)